src/components/pagination/pagination.scss

```diff
@@ -0,0 +1,265 @@
+@use '../../design-tokens/colors.yml';
+@use '../../design-tokens/font.yml';
+@use '../../design-tokens/motion.yml';
+@use '../../design-tokens/sizes.yml';
+@use '../../mixins/focus';
+@use '../../mixins/ms';
+@use '../../mixins/unit';
+
+// Defines the hit area for actionable elements, specifically the height of
+// links and the width/height of "previous" and "next" affordances.
+$action-size: sizes.$control-height;
+
+// Amount of padding within action elements. Used to minimize visual competition
+// between actions and direction affordances.
+$action-pad: ms.step(-6);
+
+// Default dimensions for a small affordance used to indicate different states
+// of individual pages. The dimensions change slightly for the current state
+// indicator. We use edge tokens because the indicator feels similar in visual
+// appearance to a border or outline.
+$indicator-height: sizes.$edge-medium;
+$indicator-width: ms.step(-1);
+$indicator-height-current: sizes.$edge-large;
+$indicator-width-current: $indicator-height-current;
+
+/**
+ * We have no styles to apply to the root `c-pagination` class, but we don't
+ * want to encourage usage of this component _without_ a containing `<nav>`
+ * element for accessibility, so we kick off our styles one element deep.
+ * Perhaps in the future the extra containing element will come in handy for
+ * modifiers and such?
+ */
+
+/**
+ * List containing page links
+ *
+ * 1. Arrange children horizontally.
+ * 2. Override default list styles. It's important that this element have
+ *    `role="list"` set to avoid accessibility issues in Safari/VoiceOver.
+ * 3. Setting this allows us to position "previous" and "next" affordances
+ *    of individual items relative to this container.
+ *
+ * @see https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html
+ */
+
+.c-pagination__items {
+  display: flex; /* 1 */
+  list-style: none; /* 2 */
+  padding-left: 0; /* 2 */
+  position: relative; /* 3 */
+}
+
+/**
+ * We initialize the list's pseudo elements so they will reserve space for the
+ * "previous" and "next" affordances.
+ */
+
+.c-pagination__items::before,
```

Comment:
Because `c-pagination__item` doesn't need to be nested, and it felt weird nesting this block and then _not_ nesting the next (or repeating the `flex: 100%` in two different places).